presto-accumulo/src/main/java/com/facebook/presto/accumulo/conf/AccumuloConfig.java

```diff
@@ -0,0 +1,333 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.accumulo.conf;
+
+import com.facebook.presto.accumulo.metadata.AccumuloMetadataManager;
+import com.facebook.presto.spi.PrestoException;
+import io.airlift.configuration.Config;
+import org.apache.commons.configuration.ConfigurationException;
+import org.apache.commons.configuration.PropertiesConfiguration;
+
+import javax.validation.constraints.NotNull;
+
+import java.io.File;
+import java.net.URL;
+
+import static com.facebook.presto.accumulo.AccumuloErrorCode.VALIDATION;
+import static java.lang.String.format;
+
+/**
+ * File-based configuration properties for the Accumulo connector
+ */
+public class AccumuloConfig
+{
+    public static final String INSTANCE = "instance";
+    public static final String ZOOKEEPERS = "zookeepers";
+    public static final String USERNAME = "username";
+    public static final String PASSWORD = "password";
+    public static final String ZOOKEEPER_METADATA_ROOT = "zookeeper.metadata.root";
+    public static final String METADATA_MANAGER_CLASS = "metadata.manager.class";
+    public static final String CARDINALITY_CACHE_SIZE = "cardinality.cache.size";
+    public static final String CARDINALITY_CACHE_EXPIRE_SECONDS =
+            "cardinality.cache.expire.seconds";
+    public static final String MINI_ACCUMULO_CLUSTER =
+            "mini.accumulo.cluster";
+
+    private String instance;
+    private String zooKeepers;
+    private String username;
+    private String password;
+    private String zkMetadataRoot;
+    private String metaManClass;
+    private Integer cardinalityCacheSize;
+    private Integer cardinalityCacheExpireSeconds;
+    private Boolean isMiniAccumuloCluster;
+
+    /**
+     * Gets the Accumulo instance name
+     *
+     * @return Accumulo instance name
+     */
+    @NotNull
+    public String getInstance()
+    {
+        return this.instance;
+    }
+
+    /**
+     * Sets the Accumulo instance name
+     *
+     * @param instance Accumulo instance name
+     * @return this, for chaining
+     */
+    @Config(INSTANCE)
+    public AccumuloConfig setInstance(String instance)
+    {
+        this.instance = instance;
+        return this;
+    }
+
+    /**
+     * Gets the ZooKeeper quorum connect string
+     *
+     * @return ZooKeeper connect string
+     */
+    @NotNull
+    public String getZooKeepers()
+    {
+        return this.zooKeepers;
+    }
+
+    /**
+     * Sets the ZooKeeper quorum connect string
```

Comment:
Don't you want to use Airlift's `@ConfigDescription` annotation instead of javadoc comment?